### PR TITLE
feat(test): Miscellaneous improvements to `test_tasks.js`

### PR DESCRIPTION
**Pull request by cpcallen**
## The basics

- [X] I branched from develop
- [X] My pull request is against develop
- [X] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)
- [X] I ran `npm run format` and `npm run lint`

### Proposed Changes

* **feat**: Modify `runTestBlock` so that it can run any async task, not just ones that return a `Promise`, by using the `async-done` package (part of Gulp, and already an indirect dependency) to detect task completion.  Celebrate by renaming it to runTestTask.
* Create a `Tester` class to encapsulate the `runTestTask`, `reportTestResult` and `runAll` functions.
* Remove the `--silent` flag from npm scripts so as not to suppress syntax error in gulpfiles.
* Have the `test` task invoke the `buildAdvancedCompilationTest` task (via `onlyBuildAdvancedCompilationTest`, to skip already-run prerequisites) directly, rather than by running `npm`.

### Reason for Changes

* Generally cleaner code.
* Remove npm scripts that aren't intended to be used by humans.

### CODE REVIEW

File: scripts/gulpfiles/test_tasks.js [hunk 1]
```diff
@@ -32,48 +32,84 @@ const BOLD_GREEN = '\x1b[1;32m';
 const BOLD_RED = '\x1b[1;31m';
 const ANSI_RESET = '\x1b[0m';
 
-let failerCount = 0;
+class Tester {
```
Comment (BeksOmega) [inline scripts/gulpfiles/test_tasks.js:35]: What's the advantage of encapsulating these in a class?

File: package.json [hunk 2]
```diff
@@ -41,7 +41,6 @@
     "lint:fix": "eslint . --fix",
     "langfiles": "gulp langfiles",
     "minify": "gulp minify",
-    "only:compile:advanced": "gulp onlyBuildAdvancedCompilationTest --debug",
```
Comment (BeksOmega) [inline package.json:44]: Yay, goodbye!

**Review by BeksOmega — COMMENTED** [review]
Slightly confused about the decision to make a Tester class. Otherwise generally LGTM, but want to understand that before approving.

File: scripts/gulpfiles/test_tasks.js (hunk 1 above)
Comment (cpcallen) [reply scripts/gulpfiles/test_tasks.js:35]: Mainly just to remove the `failerCount` global variable, and allow us to also count successes without adding a second global.

**Comment by cpcallen** [comment]
There were merge conflicts, and it seems like I forgot a needed `require` from `test_tasks.js` (did I not test this locally?  I'm _sure_ I did, and yet…) so I did a rebase and force-push.

Just awaiting approval, assuming you think my justification for the `Tester` class is sufficient.

**Pull request merged**